doozerlib/image.py

```diff
@@ -426,31 +420,21 @@ def does_image_need_change(self, changing_rpm_packages=None, buildroot_tag=None,
                         # is not about to change. Ignore the installed package.
                         self.logger.debug(f'Found latest assembly specific build ({latest_assembly_build_nvr}) for group package {package_name} is already installed in {dgk} archive; no tagging change search will occur')
                         continue
-
-                    # Several RPMs may belong to the same package, and each archive must use the same
-                    # build of a package, so all we need to collect is the set of build_ids for the packages
-                    # across all of the archives.
-                    builds_contained_in_archives[build_id] = build
-
-        self.logger.info(f'Checking whether any of the installed builds {len(builds_contained_in_archives)} has been tagged by a relevant tag since this image\'s build brew event {image_build_event_id}')
-
-        installed_builds = list(builds_contained_in_archives.values())
-        # Shuffle the builds before starting the threads. The reason is that multiple images are going to be performing
-        # these queries simultaneously. Those images have similar packages (typically rooting in a rhel base image).
-        # The KojiWrapper caching mechanism will allow two simultaneous calls to a Koji API to hit the actual
-        # server since no result has yet been returned. Shuffling the installed package list spreads the threads
-        # out among the packages to reduce re-work by the server.
-        random.shuffle(installed_builds)
-        changes_res = runtime.parallel_exec(
-            f=lambda installed_package_build, terminate_event: is_image_older_than_package_build_tagging(self, image_build_event_id, installed_package_build, newest_image_event_ts, oldest_image_event_ts),
-            args=installed_builds,
-            n_threads=10
-        )
-
-        for rebuild_hint in changes_res.get():
-            if rebuild_hint.rebuild:
-                return self, rebuild_hint
-
+                    # Add this rpm_entry to arch_rpms in order to chech whether it is latest in repos
+                    arch_rpms[arch].append(rpm_entry)
+
+        self.logger.info(f'Checking whether any of the installed rpms {len(arch_rpms)} is outdated')
+        loop = asyncio.new_event_loop()
```

Comment:
calling `asyncio.new_event_loop()` within a synchronous function has shown to be problematic in the past. I think we should make `does_image_need_change` async, and with it `config_scan_source_changes` that calls it. We might even replace current [runtime.parallel_exec](https://github.com/openshift-eng/doozer/blob/master/doozerlib/cli/scan_sources.py#L172) with a cleaner `asyncio.gather` if feasible, but since we're quering koji API this might make the process slower. It might even be a nice chance to look into how to wrap those synchronous API and make them async